kernel/src/circular_queue_kern.c

```diff
@@ -215,23 +215,20 @@ int osquery_cqueue_dropped_data(osquery_cqueue_t *queue) {
   return drops;
 }
 
-void *osquery_cqueue_reserve(osquery_cqueue_t *queue, osquery_event_t event) {
+void *osquery_cqueue_reserve(osquery_cqueue_t *queue,
+                             osquery_event_t event,
+                             size_t size) {
   void *ret = NULL;
   lck_spin_lock(queue->lck);
   if (!queue->initialized) {
     ret = NULL;
     goto error_exit;
   }
-  
+
   osquery_data_header_t *header = NULL;
 
-  ssize_t event_size = osquery_sizeof_event(event);
-  size_t size = event_size;
-  if (event_size >= 0) {
-    // Event exist.  If the event doesnt exist we will try to allocate
-    // a too large amount of space and the allocation will fail.
-    size += sizeof(osquery_data_header_t);
-  }
+  size_t contents_size = size;
+  size += sizeof(osquery_data_header_t);
```

Comment:
A call to `osquery_cqueue_reserve` will add a `sizeof(...process_t) + ...`. Meaning the caller is still responsible for including the size of it's custom/specific struct. The `sizeof(osquery_data_header_t)` is then added.